admin-dev/themes/new-theme/js/pages/product/services/paginated-combinations-service.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import Router from '@components/router';
+
+const {$} = window;
+
+export default class PaginatedCombinationsService {
```

Comment:
copied all the code from previous version of CombinationService, thats why this isn't `.ts` yet. I don't want to increase PR scope more as it wouldn't be very much related converting it to typescript.

Reply:
As typescript is now merged, you introduce a new file, you need to use TypeScript, it's completely related 😆

Reply:
I agree now that everything has been migrated to typescript let's not integrate back some JS code